authority/tls.go

```diff
@@ -132,14 +134,46 @@ func (a *Authority) Sign(csr *x509.CertificateRequest, signOpts provisioner.Sign
 
 		// Extra information from ACME attestations.
 		case provisioner.AttestationData:
-			attData = k
-			// TODO(mariano,areed): remove me once attData is used.
-			_ = attData
+			attData = &k
+
+		// Capture the provisioner's webhook controller
+		case webhookController:
+			webhookCtl = k
+
 		default:
 			return nil, errs.InternalServer("authority.Sign; invalid extra option type %T", append([]interface{}{k}, opts...)...)
 		}
 	}
 
+	// Call enriching webhooks
+	if webhookCtl != nil {
+		var attested *webhook.AttestationData
+		if attData != nil {
+			attested = &webhook.AttestationData{
+				PermanentIdentifier: attData.PermanentIdentifier,
+			}
+		}
+		whEnrichReq, err := webhook.NewRequestBody(
+			webhook.WithX509CertificateRequest(csr),
+			webhook.WithAttestationData(attested),
+		)
```

Comment:
The token should be part of the body too.

Reply:
10399d92